Validator/FormatValidator.php

```diff
@@ -50,8 +50,16 @@ public function validate($value, Constraint $constraint)
             }
         }
 
-        if (!array_key_exists($value->getDefaultFormat(), $formats)) {
-            $this->context->addViolation('invalid format');
+        $galleryDefaultFormat = $value->getDefaultFormat();
+
+        if ($formats) {
+            if (!array_key_exists($galleryDefaultFormat, $formats) && $galleryDefaultFormat !== 'reference') {
+                $this->context->addViolation('invalid format');
+            }
+        } else {
+            if ($galleryDefaultFormat !== 'reference') {
```

Comment:
Yes, but it also increases number of possible paths a.k.a. cyclomatic complexity. If we ignore the condition line 56 there are Only 3 possible paths with `elseif` : inside the first block, inside the second, inside none of them. with this version however, you can go inside the first block, inside the second but not inside the nested if, inside the second and inside the nested if, and inside none of them, which makes 4 possible paths. In the end, it makes it less easy to reason about. Also, I notice that you test for `$galleryDefaultFormat !== 'reference'` in both conditions. Maybe you could rewrite this like this : 


```php
if ($galleryDefaultFormat !== 'reference') {
   if (!$formats || !array_key_exists($galleryDefaultFormat, $formats)) {
       $this->context->addViolation('invalid format');
   }
}
```

Which simplifies into 

```php
if ($galleryDefaultFormat !== 'reference' 
   && !($formats && array_key_exists($galleryDefaultFormat, $formats))) {
       $this->context->addViolation('invalid format');
   }
}
```

Only 2 possible paths in the end.